hamilton/graph.py

```diff
@@ -51,18 +51,45 @@ def add_dependency(
     """
     if param_name in nodes:
         # validate types match
-        required_node = nodes[param_name]
-        if not types_match(adapter, param_type, required_node.type):
+        required_node: Node = nodes[param_name]
```

Comment:
Type hint necessary? Should be a `dict[str, Node]`

Reply:
No, it was useful for pycharm so I could debug it. Otherwise it is a node object, not a dict.